lib/Driver/ToolChains.cpp

```diff
@@ -37,8 +38,52 @@ using namespace swift;
 using namespace swift::driver;
 using namespace llvm::opt;
 
-/// The limit for passing a list of files on the command line.
-static const size_t TOO_MANY_FILES = 128;
+bool ToolChain::JobContext::shouldUseInputFileList() const {
+  if (Args.hasArg(options::OPT_driver_use_filelists))
+    return true;
+  return getTopLevelInputFiles().size() > TOO_MANY_FILES;
+}
+
+bool ToolChain::JobContext::shouldUsePrimaryInputFileList() const {
+  // SingleCompile's must not return true because then all inputs erroneously
+  // end up in primary file list.
+  if (Args.hasArg(options::OPT_driver_use_filelists))
+    return true;
+  return InputActions.size() > TOO_MANY_FILES;
+}
+
+bool ToolChain::JobContext::shouldFilterInputsByType() const {
+  // SingleCompile's must not filter inputs by type in order for the
+  // playground (test) to work.
```

Comment:
Still don't think this is a good idea, because it's not the real condition.

Reply:
If I remove this test, then WMO compiles will behave differently than before. This test will fail: SourceKit_CodeComplete/complete_playground_symlink_swift, but no other smoke tests, I believe. I agree that it's disturbing to break symmetry by not screening inputs for WMO compiles.  Do you think I should go to whomever would know about how playgrounds use the compiler and ask? Or what would you like me to do? (Maybe I'm misunderstanding your thoughts here.)

Reply:
(We went and talked to Ben Langmuir about this and discovered that the current behavior is…well, sort of relied upon. I asked David to change the comment to say something more like 'SingleCompile used to behave like this, and some clients are relying on this (i.e. they pass inputs that don't have ".swift" as an extension)'.)

Reply:
I've added a comment in a pidgin of Dave-ese and Jordan-ese. Can change it if you like.